server/consumer.go

```diff
@@ -4373,3 +4607,12 @@ func (o *consumer) processStreamSignal(_ *subscription, _ *client, _ *Account, s
 		o.signalNewMessages()
 	}
 }
+
+type filteredSubject struct {
```

Comment:
Having type definitions at the bottom of the file is a bit weird. Either put it on the top of the file or near where it's used.